### PR TITLE
Remove method overloading complexity

### PR DESCRIPTION
**Pull request by nielsm5**

### CODE REVIEW

**Review by tnleeuw — CHANGES_REQUESTED** [review]
Nice improvement.
I think that if you also update the class DB2XMLWriter, you should be able to get rid off the overloaded `toString()` method in `SaxElementBuilder` altogether.

**Comment by nielsm5** [comment]
> Nice improvement. I think that if you also update the class DB2XMLWriter, you should be able to get rid off the overloaded `toString()` method in `SaxElementBuilder` altogether.

Yea, at this point I was happy everything was still working. But it definitely requires a few more iterations!
Do you want me to continue in this branch or in a new pr?

**Comment by tnleeuw** [comment]
> > Nice improvement. I think that if you also update the class DB2XMLWriter, you should be able to get rid off the overloaded `toString()` method in `SaxElementBuilder` altogether.
> 
> Yea, at this point I was happy everything was still working. But it definitely requires a few more iterations! Do you want me to continue in this branch or in a new pr?

I think that there's only one use left of the `SaxElementBuilder.toString()` so I think it's worth removing that, and the overloaded `toString()`, in this PR.
Other changes and iterations can come in a new PR.

**Review by tnleeuw — APPROVED** [review]

File: core/src/main/java/org/frankframework/documentbuilder/XmlDocumentBuilder.java [hunk 1]
```diff
@@ -27,12 +28,8 @@ public class XmlDocumentBuilder extends XmlNodeBuilder implements IDocumentBuild
 	private Writer writer;
 	private SaxDocumentBuilder saxDocumentBuilder;
 
-	public XmlDocumentBuilder(String rootElement) throws SAXException {
-		this(rootElement, new StringWriter(), true);
-	}
-
 	public XmlDocumentBuilder(String rootElement, Writer writer, boolean prettyPrint) throws SAXException {
-		this(new SaxDocumentBuilder(null, writer), rootElement);
+		this(new SaxDocumentBuilder(null, new XmlWriter(writer), true), rootElement);
```
Comment (evandongen) [inline core/src/main/java/org/frankframework/documentbuilder/XmlDocumentBuilder.java:32]:
```suggestion
		this(new SaxDocumentBuilder(null, new XmlWriter(writer), prettyPrint), rootElement);
```

?

**Review by evandongen — APPROVED** [review]

**Pull request merged**